app/models/group.rb

```diff
@@ -1,3 +1,14 @@
 class Group < ApplicationRecord
   validates :name, presence: true
+  before_create :set_external_id
```

Comment:
This commit takes the decision that the random string which will be used to reference the object from outside the system will be generated in the application code rather than the DB.

It's possible to use postgres functions (and maybe the pg_crypto add on) to generate default values when items are inserted into the DB. Pushing the logic into the DB means that other users of the DB such as rails console or direct access through SQL and AWS would ensure sensible id's are always generated.

The disadvantage is that it's slightly less obvious to application developers how this is happening as we don't often use DB for functions. The same is true for testing and I think it would be harder to make changes in the future.

Instead we generate the values in ruby in the application and enforce the unique constraint at the database level. This makes it clear how the value is generated, gives us flexibility in the future to do other things but stops duplicates entering the db.